CheckPointHEC Pack first version

chkp-christiang opened the pull request:
## Contributing to Cortex XSOAR Content
New pack CheckPointHEC to integrate with Check Point Harmony Email & Collaboration 

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
No issues, new pack

## Description
New pack to integrate with Check Point Harmony Email & Collaboration Smart API

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=27861) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=27861) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=27861) it.</sub>

kgal-pan commented on Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.py:
```diff
@@ -0,0 +1,175 @@
+    raise Exception(f'Entity with id {entity} not found')
+
+
+def main() -> None:
```
Let's add this to ignore `main` method coverage since it [we're failing coverage](https://circleci.com/gh/demisto/content/708331):

```suggestion
def main() -> None:  # pragma: no cover
```

You can use the [coverage report](https://output.circle-artifacts.com/output/job/50f16fe8-9370-4fb6-b135-ea8fb24d74c9/artifacts/0/artifacts/coverage_report/html/d_4fc29e359444d752_CheckPointHEC_py.html) to see what lines are being hit/missed.

kgal-pan commented on Packs/CheckPointHEC/pack_metadata.json:
```diff
@@ -0,0 +1,27 @@
+    "categories": [
+        "Data Enrichment & Threat Intelligence"
+    ],
+    "tags": [],
```
```suggestion
    "tags": ["Email", "Security"],
```

kgal-pan commented on Packs/CheckPointHEC/pack_metadata.json:
```diff
@@ -0,0 +1,27 @@
+        "Data Enrichment & Threat Intelligence"
+    ],
+    "tags": [],
+    "useCases": [],
```
```suggestion
    "useCases": ["Email"],
```

kgal-pan replied:
Sorry about that! I meant to add it to categories

kgal-pan commented on Packs/CheckPointHEC/IncidentTypes/incidenttype-CheckPointHEC_Security_Event.json:
```diff
@@ -0,0 +1,28 @@
+    "version": -1,
+    "vcShouldIgnore": false,
+    "locked": false,
+    "name": "CheckPointHEC Security Event",
```
You might want to add a custom incident layout for this incident type which includes all the new custom incident fields introduced. See [this article](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.8/Cortex-XSOAR-Administrator-Guide/Customize-Incident-Layouts) with an explanation.

kgal-pan commented on Packs/CheckPointHEC/pack_metadata.json:
```diff
@@ -0,0 +1,27 @@
+    "url": "https://supportcenter.checkpoint.com/",
+    "email": "EmailSecurity_Support@checkpoint.com",
+    "categories": [
+        "Data Enrichment & Threat Intelligence"
```
```suggestion
        "Email"
```

chkp-christiang commented:
@kgal-pan Not sure why the validation is failing in CircleCI

kgal-pan commented:
> @kgal-pan Not sure why the validation is failing in CircleCI

Looks like the container got killed (error code 137), probably due to OOM.
I reran the workflow, let's see if it resolves itself.

kgal-pan commented:
> > @kgal-pan Not sure why the validation is failing in CircleCI
> 
> Looks like the container got killed (error code 137), probably due to OOM. I reran the workflow, let's see if it resolves itself.

@chkp-christiang - Looks like the outage has been resolved.

mmhw commented on Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.yml:
```diff
@@ -0,0 +1,209 @@
+  additionalinfo: Comma-separated list of types of alerts to fetch. Types might change over time. Some examples are 'Bug' and 'Vulnerability'
+- section: Collect
+  display: Minimum severity of alerts to fetch
+  name: min_severity
```
It seems that the min_severity param is not used in the script, please go through the yml file and delete unnecessary params (if there are).

chkp-christiang replied:
@mmhw The problem with those params is that validation does not pass if they are missing.

mmhw replied:
Hi @chkp-christiang,
Can you please share the error, I can't reproduce it.

mmhw reviewed (commented):
Hi @chkp-christiang,
Thank you for your contribution!
Good work :)

Please note my comments.
Please also delete the files Pipfile and Pipfile.lock.

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

mmhw commented on Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.py:
```diff
@@ -97,9 +97,17 @@ def test_module(client: Client):
 
-def fetch_incidents(client: Client):
+def fetch_incidents(client: Client, first_fetch: str):
+    def get_timedelta() -> timedelta:
```
Good work!
You can use the `parse_date_range` function to convert the `first_fetch` to a time.
see [here](https://github.com/demisto/content/blob/master/Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py#L1254).

kgal-pan commented:
> @mmhw **Infrastructure testing** failing in Unit Tests.

@chkp-christiang - This will be fixed by https://github.com/demisto/content/pull/28033. Once it's merged, please pull from `master`.

chkp-christiang commented:
> > @mmhw **Infrastructure testing** failing in Unit Tests.
> 
> @chkp-christiang - This will be fixed by #28033. Once it's merged, please pull from `master`.

Thanks @mmhw, 🟢 now.

efelmandar reviewed (commented):
Hi @chkp-christiang, 
Thank you for your contribution! I appreciate the effort you put into this PR. Overall, the content is looking good, but I have a few points that I want to address before we can merge this pull request. Please take a look at the following suggestions:

1. **Use `State` incident field from Common**: Instead of creating a new `CheckPointHEC State` field, it would be more appropriate to utilize the existing `State` field from the CommonTypes pack. This will help maintain consistency and avoid unnecessary duplication. Please update the code accordingly.

2. **Improve incident fields description**: The description of the incident fields could be enhanced to provide more clarity. For example, it would be helpful to specify the type and state of what exactly. This will ensure a better understanding of the data contained within these fields. Please revise the incident field descriptions accordingly.

3. **No `auto extract indicators` for incident type**: Currently, there are no auto-extract indicators defined for the incident type. If this is intended I would consider maybe adding extracting indicators in the playbook to enrich the incident for the analyst.

4. **No default playbook set for incident type**: It seems that there is no default playbook set for the incident type. It would be helpful to have a default playbook defined to handle these incidents automatically. Could you please add a default playbook for the incident type?

5. **Purpose of the PB with only 1 task**: Usually playbooks with a single task are not recommended and should be used only in specific use cases. Could you please elaborate on the purpose of the playbook? in relation to the previous point, if this is a default playbook to run on each incident consider setting it as `Default Playbook` in the incident type configuration.

If you have any questions or need further help, feel free to ask here or via DFIR workspace on Slack.

Once we have addressed these points we will be able to proceed with the review process and merge of this PR. Keep up the good work!

chkp-christiang commented:
Hi @efelmandar,

Thanks for your comments, I implemented some changes and want to clarify others that were not implemented:

1.  The field `State` from `CommonTypes` pack has been implemented instead of the `CheckPointHEC State`, this last has been also deleted as not need anymore, the layout has been updated accordingly to use the new added field.

2. Custom incident field descriptions have been improved.

3. We are in control of what's coming back as a result the API call, the data is parsed well to our need and extracting data from text is not needed at this point.

4. In most cases it's enough for the admin to see the first response and the second call is not often needed, only when the admin reads the security event repost be may realize that more info is needed, and only then he may manually want to call the second step.

5. CheckPoint Harmony Email and Collaboration system is scanning emails in order to protect customers from phishing and malware attacks, the XSOAR solution is useful for admins to get notifications upon detecting such attacks and analyze them further. The admin can then use CheckPoint system to take action, for example, to quarantine the email, these actions are planned to be supported by our XSOAR solution in the near future

mmhw commented:
Hi @chkp-christiang,

For me the PR is good.
Just pay attention to update the README file, following the deletion of the unnecessary params - `demisto-sdk generate-docs -i Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.yml`.

We will wait for confirmation from @efelmandar, then we will schedule a time for a demo, or you will send a video demonstrating it.

chkp-christiang commented:
> Hi @chkp-christiang,
> 
> For me the PR is good. Just pay attention to update the README file, following the deletion of the unnecessary params - `demisto-sdk generate-docs -i Packs/CheckPointHEC/Integrations/CheckPointHEC/CheckPointHEC.yml`.
> 
> We will wait for confirmation from @efelmandar, then we will schedule a time for a demo, or you will send a video demonstrating it.

Hi @mmhw,

Thanks for pointing out the docs, really appreciate all your help, I'm trying to prepare a demo video asap.

chkp-christiang commented:
@mmhw @efelmandar We have recorded this video, please let us know for further questions: https://drive.google.com/file/d/1LIL63t06GV9uZncwiFxHQ8aYR-YgN-GO/view?usp=sharing

mmhw commented:
> 

Thanks for the video,
Sorry for not detailing what I expect to see, I would also like to see the integration instance creation process, to see that the test button works, as well as to see the error it throws, when there is a url or the Client Secret is incorrect.

And if you can show an example of running the `checkpointhec-get-entity` command, and show that the outputs are as they should be.

Thanks again

efelmandar commented:
@chkp-christiang Thank you very much for your detailed response and video recording it explains the use case and workflow of the analyst when using your content, which makes more sense to the choices you've made regarding points 3,4 in my review.
I do want to suggest taking advantage of the layout and populating incident fields to better present the information to the analyst, instead of just showing it in the war room - but as you said, this and other things can be done in a later stage.

Overall the PR looks good and is approved from my side - @mmhw FYI

@chkp-christiang Thank you again for your contribution and hard work, I'm looking forward to seeing this pack evolve.

mmhw reviewed (approved):
Hi @chkp-christiang,
Really good work :)

mmhw commented:
> @mmhw here is the link with the new video: https://drive.google.com/file/d/1myDeX5M7JQkU-7fbC1yJWInqg5nUnexC/view?usp=sharing

Thanks, that's great!
